nulib/meadow
# Adds GraphQL query support for picking s3 objects in the ingest bucket

# Summary 
Users can now choose between drag-and-drop or an s3 object picker for adding and replacing file sets in the UI.

# Specific Changes in this PR
- Adds GraphQL query support for picking s3 objects in the ingest bucket
- Backend implemented with a new S3 backend utility module.
- Defines a `S3ObjectPicker` component, now used by the file set modal and file set replace modal. The `S3ObjectPicker` automatically filters the objects based on their `mime_type` (gathered via head requests), while reusing existing mime type validator code.

# Version bump required by the PR

See [Semantic Versioning 2.0.0](https://semver.org/) for help discerning which is required.


# Steps to Test
- Spin up Meadow in your dev environment and add a work.
- Add a few files to your ingest bucket in s3 that matches the mime type of your work. Files that will be replacements need to be uploaded under the `file_sets/` prefix in the ingest bucket.
- Go to the preservation tab and hit the button to add a file set.
- Pick the file from the s3 object picker and fill out the form to ingest it
- Go the preservation tab and select the "replace file set" option in the dropdown menu for the file set.
- When the modal pops up replace the file set with another one from the ingest bucket and submit the form.
- File set should be replaced in the background shortly (can check the action states in the preservation tab dropdown to check)

# :rocket: Deployment Notes

**Note** - __if you check any of these boxes go to the [(always open) `main` <- `staging` PR](https://github.com/nulib/meadow/pulls) and add detailed notes and instructions to help out others who may end up deploying your changes to production__

- Backward compatible API changes
- Backwards-incompatible API changes


# Tested/Verified
